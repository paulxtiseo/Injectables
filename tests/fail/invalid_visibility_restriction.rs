use injectables::{injectable, inject_fields};

mod inner {
  use super::*;

  #[injectable]
  pub struct Restricted {
    pub(in crate::invalid::path) field: String,  // invalid visibility path
  }
}

#[inject_fields(inner::Restricted)]
pub struct InvalidVisibility {
  pub name: String,
}

fn main() {}