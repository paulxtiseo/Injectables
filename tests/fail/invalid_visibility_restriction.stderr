error[E0433]: failed to resolve: could not find `invalid` in the crate root
 --> tests/fail/invalid_visibility_restriction.rs:8:19
  |
8 |     pub(in crate::invalid::path) field: String,  // invalid visibility path
  |                   ^^^^^^^ could not find `invalid` in the crate root

error: Cannot access field 'field' with visibility Restricted("crate :: invalid :: path") from module 'inner' in module ''
  --> tests/fail/invalid_visibility_restriction.rs:12:1
   |
12 | #[inject_fields(inner::Restricted)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `inject_fields` (in Nightly builds, run with -Z macro-backtrace for more info)
